src/linux/sentry_procmaps_modulefinder.c

```diff
@@ -206,24 +216,33 @@ load_modules(void)
     char buf[4096];
     sentry_stringbuilder_t sb;
     sentry__stringbuilder_init(&sb);
-    while (1) {
+    while (true) {
         ssize_t n = read(fd, buf, 4096);
         if (n <= 0) {
             if (errno == EAGAIN || errno == EINTR) {
                 continue;
             }
             break;
         }
-        sentry__stringbuilder_append_buf(&sb, buf, n);
+        if (sentry__stringbuilder_append_buf(&sb, buf, n)) {
+            close(fd);
+            goto done;
```

Comment:
Instead of all those `goto`s, how about we return the number of modules read, and take care of freezing, logs etc in the caller?